[NETBEANS-5656] Various smaller fixes in the database New Connection wizard

eirikbakke opened the pull request:
Various small UI improvements to the Database module's "New Connection" wizard:
- Refactor the wizard to avoid depending on hardcoded wizard panel indexes.
- Improve two error messages in "New Connection" wizard.
- Avoid a spurious 'Specified class is not a driver' warning in certain situations.
- When detecting the Microsoft SQL Server JDBC driver, avoid showing an ancient version number.
- Add a default port number for Microsoft SQL Server.
- Tidy up presentation names in the JDBC URL format dropdown.
- Decrease the timeout of the connection validation step.

matthiasblaesing reviewed (commented):
Some of the wording change could be discussed, for example replacing braces with a slash because there are already braces, misses IMHO the point, that there are drivers with slashes in the name, making it possible to return to the original behavior.

I left two inline comments.

eirikbakke commented on ide/db/src/org/netbeans/modules/db/explorer/Bundle.properties:
```diff
@@ -21,7 +21,10 @@ SchemaIsNotSet=Default schema
-EXC_CannotEstablishConnection=Cannot establish a connection to {0} using {1} ({2})
+# The "New Connection Wizard" only has space for the first part of the message, so put the message
+# from the driver (parameter {2}), which usually explains the error, first.
+#EXC_CannotEstablishConnection=Cannot establish a connection to {0} using {1} ({2})
```
Sure.

eirikbakke commented on ide/db/src/org/netbeans/modules/db/util/Bundle.properties:
```diff
@@ -37,6 +37,7 @@ DRIVERNAME_OracleOCI=Oracle OCI
 DRIVERNAME_JTDS=jTDS
 DRIVERNAME_DB2JCC=IBM DB2 Universal Driver
 DRIVERNAME_MSSQL2005=Microsoft SQL Server 2005
+DRIVERNAME_MSSQL=Microsoft SQL Server
```
OK, I'll remove the DRIVERNAME_MSSQL2005 key, since it includes the no-longer-relevant "2005" year.

eirikbakke replied:
Grepped for this, found no other uses of DRIVERNAME_MSSQL2005; removing that key as suggested.

eirikbakke commented:
I'm fine with taking out the slash vs. parentheses thing. Depends on how much you think of this as a "presentation name" vs. a string you should be able to parse (even mentally) back to its original.

eirikbakke commented:
Reverted the "replacing braces with a slash" part (and removed mention of it from the commit message), and applied the other two suggestions from the in-code comments.

matthiasblaesing reviewed (approved):
Looks sane to me - lets get this in.

eirikbakke commented:
Thanks!

matthiasblaesing commented:
@eirikbakke for your information: this PR broke unittests - this should fix it: #2975

eirikbakke commented:
Oh, sorry! I thought I had waited until all the tests had passed... thank you!
